build: add `release` as a valid commit message subject

matsko opened the pull request:
This should make quiet down the commit message CircleCI issues.

matsko committed:
build: add `release` as a valid commit message subject

gkalpak commented:
It will silece CircleCI (for now), but is a work-around, not a proper fix.
The reason `validate-commit-message` complains for a commit that is not part of the PR is tries to validate, is that it calculates the commit range incorrectly.

Background:
`package.json#version` is being used by `validate-commit-message` to determine the branch a PR is submitted against. Right now it thinks PR for master are actually targeting the 5.x branch and it tries to validate more commits than necessary (and fails due to the bad `release: ...` commit).

One possible fix (now that there is a 5.x branch) is to bump `package.json#version` to `6.something`.

(If for some reason it is not desirable to do so until a 6.x version is released, we need to find a different way to identify a PR's target branch.)

gkalpak commented on tools/validate-commit-message/commit-message.json:
```diff
@@ -28,6 +28,7 @@
     "platform-server",
     "platform-webworker",
     "platform-webworker-dynamic",
+    "release",
```
If you really want to add this as valid commit message subject, you must update the corresponding tests as well.

alxhub replied:
And CONTRIBUTORS.md

matsko replied:
No I don't think we should make this a known subject because nobody external will do anything relative to a release.

IgorMinar replied:
@matsko this is the wrong place to put this. you need to update the `types` array.

and @alxhub and @gkalpak are right, you will need to update the tests and document this even if it's not relevant for external developers. It will be relevant for anyone new joining the team and reading that doc.

matsko replied:
Damn I feel dumb. OK fixed.

matsko commented:
@gkalpak let's add the `release` subject first (in this PR) and then address the next fix on Monday.

IgorMinar commented:
all tests must be updated to contain "release" otherwise you'll have CI failures.

IgorMinar reviewed (approved):
please clean up this PR per feedback. otherwise lgtm! thanks for making this change that went unnoticed for so long.

matsko committed:
build: temporarily increase the commit msg limit to 120

Right now HEAD and 5.0.x have a branch deviation and therefore
all the commits between both branches are being compared. There
exists a problematic commit which has a commit message that is
longer than 100 commits. This patch will temporarily increase
the limit to 120 so that CI passes. Once master is resumed to
being the primary development branch (once 5.0.0 is out) then
the the msg limit will be set back to 100.

gkalpak reviewed (commented):
Still need to add `release` as a valid type in `CONTRIBUTORS.md`.
